spec_parser/owl_to_context.py

```diff
@@ -0,0 +1,112 @@
+# SPDX-FileCopyrightText: 2023 spdx contributors
+#
+# SPDX-License-Identifier: Apache-2.0
+import json
+from os import path
+
+# properties with Enum range should look like this (probably), so that their values are automatically appended
+# with the Enum URI:
+# "annotationType": {
+#     "@id": "core:annotationType",
+#     "@type": "@vocab",
+#     "@context": {
+#         "@vocab": "core:AnnotationType/"
+#     }
+# },
+
+PROPERTIES_WITH_ENUM_RANGE = [
+    "safetyRiskAssessment",
+    "sensitivePersonalInformation",
+    "annotationType",
+    "externalIdentifierType",
+    "externalReferenceType",
+    "algorithm",
+    "scope",
+    "profile",
+    "completeness",
+    "relationshipType",
+    "confidentialityLevel",
+    "datasetAvailability",
+    "decisionType",
+    "justificationType",
+    "catalogType",
+    "conditionality",
+    "sbomType",
+    "softwareLinkage",
+    "purpose",
+]
+
+REFERENCE_PROPERTY_TYPES = [
+    "core:Element",
+    "core:Agent",
+]
+
+
+def convert_spdx_owl_to_jsonld_context(spdx_owl: str, out_dir: str):
+    # spdx_owl must point to the OWL in json-ld format
+    with open(spdx_owl, "r") as infile:
+        owl_dict = json.load(infile)
+
+    context_dict = owl_dict["@context"]
+
+    for node in owl_dict["@graph"]:
+        # print(node)
+        node_type = node.get("@type")
+        if not node_type:
+            # print(node)
+            continue
+
+        if "owl:NamedIndividual" in node_type:
+            continue
+        elif node_type in ["owl:DatatypeProperty", "owl:ObjectProperty"]:
+            name = node["@id"].split(":")[-1]
```

Comment:
there should be a function to do that

like `name = get...FromId(node)`

Reply:
done